File: src/main/java/models/AudioFile.java
package models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.io.File;
import java.io.Serializable;

public class AudioFile implements Serializable {
    public String title;
    public String artist;
    public String album;
    public int track;

    public String filePath;

    @JsonIgnore
    public File file;
    public String genre;
}

File: frontend/src/app/business/domain/audio-file.ts
export class AudioFile {
  constructor(public id: number,
              public metadataRetrieved: boolean,
              public album: string,
              public artist: string,
              public path: string,
              public genre: string,
              public title: string,
              public dateChanged: Date,
              public track: number) {
  }
}
